### PR TITLE
tkuumola solution chall03

### PR DESCRIPTION
**Pull request by T7Q**
My first program in Go ..

1. Sent GET request to https://chall03.hive.fi/
2. Found matches based on regex, converted values to int and saved them to id, red, green and blue values. Created a hex from them using Sprintf.
3. Sent GET request with id and hex.

### CODE REVIEW

File: chall03/tkuumola.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	if err != nil {
+		log.Fatalln(err)
+	}
+	body, err := ioutil.ReadAll(resp.Body)
+	defer resp.Body.Close()
+	if err != nil{
+		log.Fatalln(err)
+	}
+	fmt.Println("GET	https://chall03.hive.fi/")
+	fmt.Printf("	answer: %s\n\n", string(body))
```
Comment (jgengo) [inline chall03/tkuumola.go:25]: I would recommend using \t if you want to tabulate or printf-padding.

File: chall03/tkuumola.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+	r := regexp.MustCompile(`\d+`)
```
Comment (jgengo) [inline chall03/tkuumola.go:27]: regex wasn't necessary and should be avoided for this kind of exercise: https://softwareengineering.stackexchange.com/questions/113237/when-you-should-not-use-regular-expressions

File: chall03/tkuumola.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	matches := r.FindAllString(string(body), -1)
+
+	id, err := strconv.Atoi(matches[0])
+	red, err := strconv.Atoi(matches[1])
+	green, err := strconv.Atoi(matches[2])
+	blue, err := strconv.Atoi(matches[3])
+
+	hex := fmt.Sprintf("%02x%02x%02x", red, green, blue)
+	url := fmt.Sprintf("https://chall03.hive.fi/?id=%d&resp=%s", id, hex)
```
Comment (jgengo) [inline chall03/tkuumola.go:36]: nice!

**Review by jgengo — APPROVED** [review]

**Comment by T7Q** [comment]
thank you very much for going through the code thoroughly and commenting. This helps me a lot!!!

**Pull request merged**